prow/flagutil/kubernetes_cluster_clients.go

```diff
@@ -384,20 +400,32 @@ func (o *KubernetesOptions) BuildClusterManagers(dryRun bool, callBack func(), o
 	aggregatedErr := utilerrors.NewAggregate(errs)
 
 	if aggregatedErr != nil {
-		// Retry the build clusters that failed to be connected initially, execute
-		// callback function when they become reachable later on.
-		// This is useful where a build cluster is not reachable transiently, for
-		// example API server upgrade caused connection problem.
+		// Retry the build clusters that failed to be connected initially. If
+		// suddenly we can connect to them successfully, execute the callback
+		// function (e.g., to terminate this pod to force a restart). This is
+		// useful where a build cluster is not reachable transiently, such as
+		// when an API server upgrade causes connection problems.
 		go func() {
 			for {
 				for buildClusterName, buildClusterConfig := range o.clusterConfigs {
+					// Do not check already-successfully-checked build clusters.
 					if _, ok := res[buildClusterName]; ok {
 						continue
 					}
-					if _, err := manager.New(&buildClusterConfig, options); err == nil {
-						logrus.WithField("build-cluster", buildClusterName).Info("Build cluster that failed to connect initially now worked.")
-						callBack()
+
+					// If there are any errors with this (still troublesome)
+					// build cluster, keep checking.
+					if _, err := manager.New(&buildClusterConfig, options); err != nil {
+						logrus.WithField("build-cluster", buildClusterName).Infof("failed to construct build cluster manager: %s", err)
```

Comment:
nit: Lets log this and line 423 at a lower log level (`debug` or `trace`) since this is the expected case. The useful logs are the ones indicating we initially failed to create a build cluster and the log before the `callback()` indicating a build cluster became available. Logging all the broken build clusters from every component every few seconds will be spammy.

Reply:
Done